README.md

```diff
@@ -240,11 +250,34 @@ Above `Router` object comes with the following API:
 - `query` - `Object` with the parsed query string. Defaults to `{}`
 - `push(url, as=url)` - performs a `pushState` call with the given url
 - `replace(url, as=url)` - performs a `replaceState` call with the given url
+- `ready(cb)` - fires the `cb` when router is ready (mostly used with [Router Events](#router-events))
 
 The second `as` parameter for `push` and `replace` is an optional _decoration_ of the URL. Useful if you configured custom routes on the server.
 
 _Note: in order to programmatically change the route without triggering navigation and component-fetching, use `props.url.push` and `props.url.replace` within a component_
 
+##### Router Events
+
+You can also listen to different events happening inside the Router. For that, `next/router` comes with a Node.js [EventEmitter](https://nodejs.org/api/events.html) compatible API.
+
+Supported events:
+
+- `routeChangeStart(route, url)` - Fires when a route starts to change
+- `routeChangeComplete(route, url)` - Fires when a route changed completely
+- `routeChangeError(err, route, url)` - Fires when there's an error when changing routes
+
+Here's how to property listen to a router event:
+
+```js
+Router.ready(() => {
+  Router.on('routeChangeStart', (route, url) => {
+    console.log('Route is changing to: ', route)
+  })
+})
```

Comment:
@nkzawa Check now. I made `Router.ready()` as a private method. So, user don't need to use it directly anymore.